### PR TITLE
Add VS Code tasks to build and test the C# bits

### PR DESCRIPTION
**Pull request by tamasvajk**
Added 
* tasks.json to have build and test command
* settings to disable some prompts, and set some parameters so that tests are found and they generate no build errors

### CODE REVIEW

File: csharp/.vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "recommendations": [
+        "github.vscode-codeql",
+        "ms-dotnettools.csharp"
+        // "formulahendry.dotnet-test-explorer",
```
Comment (tamasvajk) [inline csharp/.vscode/extensions.json:5]: ideally I would uncomment these, but I'm not sure of we want to recommend "random" extensions to contributors.

Comment (tamasvajk) [reply csharp/.vscode/extensions.json:5]: @calumgrant Can I add these two recommendations without being commented out?

**Comment by tamasvajk** [comment]
@calumgrant Should I prefix the commit message with C# or CSharp? I've seen that it happens for some commits.

**Comment by calumgrant** [comment]
I would normally use `C#`

File: csharp/.vscode/extensions.json (hunk 1 above)
Comment (calumgrant) [reply csharp/.vscode/extensions.json]: I'm ok with these being commented out. Normally however we don't use code for TODOs, and use GitHub Issues instead.

**Pull request merged**